connaisseur/validators/notaryv1/notary.py

```diff
@@ -89,39 +89,56 @@ def healthy(self):
         except Exception:
             return False
 
-    async def get_trust_data(
-        self,
-        session: aiohttp.ClientSession,
-        image: Image,
-        role: TUFRole,
-        token: str = None,
+    async def get_auth_and_root_trust_data(
+        self, session: aiohttp.ClientSession, image: Image, token: str = None
     ):
-        im_repo = f"{image.repository}/" if image.repository else ""
-        url = (
-            f"https://{self.host}/v2/{image.registry}/{im_repo}"
-            f"{image.name}/_trust/tuf/{str(role)}.json"
-        )
+        request_kwargs = self.__build_args(image, "root")
+
+        if token:
+            request_kwargs.update({"headers": ({"Authorization": f"Bearer {token}"})})
 
-        request_kwargs = {
-            "url": url,
-            "ssl": self.cert,
-            "headers": ({"Authorization": f"Bearer {token}"} if token else None),
-        }
         async with session.get(**request_kwargs) as response:
             status = response.status
-            if (
-                status == 401
-                and not token
-                and ("www-authenticate" in [k.lower() for k in response.headers])
+            if status == 401 and (
+                "www-authenticate" in [k.lower() for k in response.headers]
             ):
                 auth_url = self.__parse_auth(
                     {k.lower(): v for k, v in response.headers.items()}[
                         "www-authenticate"
                     ]
                 )
-                token = await self.__get_auth_token(session, auth_url)
-                return await self.get_trust_data(session, image, role, token)
+                repo_token = await self.__get_auth_token(session, auth_url)
+                root_trust_data, _ = await self.get_auth_and_root_trust_data(
+                    session, image, repo_token
```

Comment:
If something is wrong with the registry, this could allow infinite loop: 401-> getToken-> getAuth+Root -> 401 -> getToken -> getAuth+Root...., so probably best to keep the `and not token` part of the original if clause